datasets/sentinel-3/sentinel_3/sentinel_3_base.py

```diff
@@ -24,35 +32,148 @@
 }
 
 
+def recursive_round(coordinates: List[Any], precision: int) -> List[Any]:
+    """Rounds a list of numbers. The list can contain additional nested lists
+    or tuples of numbers.
+
+    Any tuples encountered will be converted to lists.
+
+    Args:
+        coordinates (List[Any]): A list of numbers, possibly containing nested
+            lists or tuples of numbers.
+        precision (int): Number of decimal places to use for rounding.
+
+    Returns:
+        List[Any]: a list (possibly nested) of numbers rounded to the given
+            precision.
+    """
+    for idx, value in enumerate(coordinates):
+        if isinstance(value, (int, float)):
+            coordinates[idx] = round(value, precision)
+        else:
+            coordinates[idx] = list(value)  # handle any tuples
+            coordinates[idx] = recursive_round(coordinates[idx], precision)
+    return coordinates
+
+
+def nano2micro(value: float) -> float:
+    """Converts nanometers to micrometers while handling floating
+    point arithmetic errors."""
+    return float(Decimal(str(value)) / Decimal("1000"))
+
+
+def hz2ghz(value: float) -> float:
+    """Converts hertz to gigahertz while handling floating point
+    arithmetic errors."""
+    return float(Decimal(str(value)) / Decimal("1000000000"))
+
+
+def crossing_longitude(
```

Comment:
Is the hope that any this will move to `antimeridian`? Or is it specific to sentinel-3?

Reply:
This is specific to sentinel-3. Perhaps we should move it to a separate `utils.py` file?

Reply:
I moved all the winding detection to a separate `sentinel_3_winding.py`.